Modules/HR/Http/Controllers/Recruitment/RecruitmentOpportunityController.php

```diff
@@ -20,12 +20,17 @@ public function getOpportunityType()
     public function index()
     {
         $this->authorize('list', Job::class);
-
-        $jobs = Job::with('applications', 'applications.applicant', 'jobRequisition')
-            ->typeRecruitment()
-            ->latest()
-            ->paginate(config('constants.pagination_size'))
-            ->appends(Request::except('page'));
+        $search = request()->query('title') ?? '';
+        if ($search != '') {
+            $jobs = Job::where('title', 'LIKE', "%$search%")
+                ->paginate(config('constants.pagination_size'));
+        } else {
+            $jobs = Job::with('applications', 'applications.applicant', 'jobRequisition')
+                ->typeRecruitment()
+                ->latest()
+                ->paginate(config('constants.pagination_size'))
+                ->appends(Request::except('page'));
+        }
```

Comment:
Why you have not added `->typeRecruitment()` and `->latest()` filters in the query when there is a search term? Any specific reason for that? 

Also instead doing it like this you can use the below suggestion. This way we can avoid duplicate code
```suggestion
        $search = request()->query('title') ?? '';
        
        $jobs = Job::with('applications', 'applications.applicant', 'jobRequisition')
            ->typeRecruitment()
            ->latest();
        
        if ($search != '') {
            $jobs = $jobs->where('title', 'LIKE', "%$search%")
                ->paginate(config('constants.pagination_size'));
        }
                
        $jobs = $jobs->paginate(config('constants.pagination_size'))
            ->appends(Request::except('page'));
```